src/main/java/com/puppycrawl/tools/checkstyle/checks/blocks/EmptyBlockCheck.java

```diff
@@ -270,23 +272,28 @@ private boolean checkIsAllLinesAreWhitespace(int lineFrom, int lineTo) {
      * @param ast a {@code DetailAST} value
      * @return the left curly corresponding to the block to be checked
      */
-    private static DetailAST findLeftCurly(DetailAST ast) {
-        final DetailAST leftCurly;
-        final DetailAST slistAST = ast.findFirstToken(TokenTypes.SLIST);
-        if ((ast.getType() == TokenTypes.LITERAL_CASE
-                || ast.getType() == TokenTypes.LITERAL_DEFAULT)
-                && ast.getNextSibling() != null
-                && ast.getNextSibling().getFirstChild() != null
-                && ast.getNextSibling().getFirstChild().getType() == TokenTypes.SLIST) {
-            leftCurly = ast.getNextSibling().getFirstChild();
+    private static Optional<DetailAST> findLeftCurly(DetailAST ast) {
+        final DetailAST parent = ast.getParent();
+        final int parentType = parent.getType();
+        final Optional<DetailAST> lcurlyOptional;
+
+        if (parentType == TokenTypes.SWITCH_RULE) {
+            lcurlyOptional = Optional.ofNullable(parent.findFirstToken(TokenTypes.SLIST));
         }
-        else if (slistAST == null) {
-            leftCurly = ast.findFirstToken(TokenTypes.LCURLY);
+        else if (parentType == TokenTypes.CASE_GROUP) {
+            final DetailAST statementList = ast.getNextSibling();
+            lcurlyOptional = Optional.ofNullable(statementList)
+                         .map(DetailAST::getFirstChild)
+                         .filter(node -> node.getType() == TokenTypes.SLIST);
         }
         else {
-            leftCurly = slistAST;
+            DetailAST leftCurly = ast.findFirstToken(TokenTypes.SLIST);
+            if (leftCurly == null) {
```

Comment:
This can be separated into another `else if`, please do this (via Optional#of since we will now have a not null check); also make a comment in each branch here about what type we expect in each.

Reply:
```
        else if (ast.findFirstToken(TokenTypes.SLIST) != null) {
            lcurlyOptional = Optional.of(ast.findFirstToken(TokenTypes.SLIST));
        }
```

Reply:
good, done